### PR TITLE
Add nameMapping option to Java client, server generators

### PR DESCRIPTION
**Pull request by wing328**
Add `nameMapping` feature to the Java client, server generator, e.g.
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g java -i modules/openapi-generator/src/test/resources/3_0/java/petstore-with-fake-endpoints-models-for-testing-okhttp-gson.yaml  -o /tmp/java2/ --name-mappings _type=underscoreType, type_=typeWithUnderscore
```
to handle the following edge case:
```
    PropertyNameCollision:
      properties:
        _type:
          type: string
        type:
          type: string
        type_:
          type: string
      type: object
```

FYI @OpenAPITools/generator-core-team 

To close https://github.com/OpenAPITools/openapi-generator/issues/9444

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/model/PropertyNameCollisionTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class PropertyNameCollisionTest {
+    private final PropertyNameCollision model = new PropertyNameCollision();
+
+    /**
+     * Model tests for PropertyNameCollision
+     */
+    @Test
+    public void testPropertyNameCollision() {
+        // TODO: test PropertyNameCollision
```
Comment (welshm) [inline samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/model/PropertyNameCollisionTest.java:36]: Intended to be done before submission?

Comment (wing328) [reply samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/model/PropertyNameCollisionTest.java:36]: I tested the new option in the java (okhttp-gson) petstore sample instead with a new schema. Will remove this file in my next PR. 

Thanks for reviewing.

Comment (wing328) [reply samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/model/PropertyNameCollisionTest.java:36]: actually this is the auto-generated test file so we will keep it. the test is added to https://github.com/OpenAPITools/openapi-generator/pull/16103/files/48c9550264c686d154be329a0984142026537e5f#diff-54b45af594b56ff301edf017e96c2786680d87f713d287fcbf5a1429072674b3R8.

the output won't compile without using the nameMapping option in Java okhttp petstore samples.

**Comment by PRTGC** [comment]
Hi...
I must be missing something..
I have a schema with a propery named "type:
    device:
        serial_no:
          type: object
          example: "null"
        type:
          type: object
          example: physical

Generating rust, and I have tried all variations, nothing works..
Please advice...
Thanks
  JR

**Comment by wing328** [comment]
please open an issue with the details (e.g. spec, full command to repeat the issue, etc)

**Pull request merged**